core/src/main/java/jenkins/MissingDependencyException.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import hudson.PluginWrapper.Dependency;
+import hudson.Util;
+
+/**
+ * Exception thrown if plugin resolution fails due to missing dependencies
+ * 
+ * @author Carlos Sanchez
+ * @since 1.650
+ *
+ */
+public class MissingDependencyException extends IOException {
+
+    private List<Dependency> missingDependencies;
+
+    private static List<String> dependenciesToString(List<Dependency> dependencies) {
+        List<String> strings = new ArrayList<>();
+        for (Dependency d : dependencies) {
+            strings.add(d.toString());
+        }
+        return strings;
+    }
+
+    public MissingDependencyException(List<Dependency> missingDependencies) {
+        super("Dependency " + Util.join(dependenciesToString(missingDependencies), ", ") + " doesn't exist");
```

Comment:
NIT: you can be missing more than one dependency...
`"One or more dependencies could not be resolved; " + Util.join...`